fast_llm/data/dataset/gpt/memmap.py

```diff
@@ -120,16 +135,53 @@ def _init(self, name: str, prefix: pathlib.Path | str, num_documents: int | None
                         offset=rejected_span_offset + idx * 2 * np.dtype(np.int32).itemsize,
                     )
                 )
+            offset += np.array(self._chosen_spans).nbytes + np.array(self._rejected_spans).nbytes
```

Comment:
This is inefficient. We can just store the image count cumsums instead. (Same for spans BTW)